### ISSUE TITLE
Need open handle check in Windows version of MVM_dir_read

### ISSUE THREAD

**Issue by zoffixznet** [MoarVM/MoarVM#598]
https://irclog.perlgeek.de/moarvm/2017-05-13#i_14576944

```
16:44	JimmyZ	timotimo:   it need too before check invalid handle
16:44		invalid hanlde != NULL
16:47		timotimo: https://github.com/MoarVM/MoarVM/blob/22fb8e49c02a8a3cae33333faa18106815c8e526/src/io/dirops.c#L297 need to add check here :)
```

**Comment by duke-m** [comment]
Hello, thanks. I don't know how the others feel but could you please rather be publishing usual bug/issue reports than pasting IRC logs? I've seen that happen quite often.

Please don't get me wrong, I don't want to break in here and tell all others how to do things, but I believe it's a matter of transparency, simplicity, efficacy and efficiency (without claiming that I do well in any of these domains :sweat_smile:). Wouldn't it be better for this project to have detailed and destilled reports so others can very quickly see if they can try to contribute with a solution? Of course adding the link to the originating chat log might be helpful, too.

Thanks and best regards,

**Comment by zoffixznet** [comment]
I copy-pasted the log, so the problem doesn't get forgotten.

Time is a premium commodity. If a [project wants me to write a dissertation](https://github.com/ugexe/zef/issues/new) for all issues, most of the time I'll just not file anything.

**Comment by ugexe** [comment]
Time is a premium commodity for the people at both ends of bug reports. No hyperbole is needed to back this up.

**Comment by zoffixznet** [comment]
Actually, I don't understand what the problem is here. The OP text shows the exact line of code where a NULL check is missing. No IRC log reading is required.

**Comment by ugexe** [comment]
The choices are not just "No issues" and "Poorly described issues" - this is not a binary thing so lets not try and make it one.

As for why someone might have questions: maybe they are familiar with [windows api](https://blogs.msdn.microsoft.com/oldnewthing/20040302-00/?p=40443/) and want to be sure there isn't a better solution. I myself cannot say for sure if NULL is valid for a windows HANDLE or not, so I am not surprised someone requested clarification.

**Comment by zoffixznet** [comment]
> The choices are not just "No issues" and "Poorly described issues" - this is not a binary thing so lets not try and make it one.

Except it is binary. I filed the best Issue I could file at the time. If you're telling me it's not good enough, the only other option is for me to not file anything at all.

I went the extra mile and posted a message one MoarVM developer sent another MoarVM developer when they were AFK. Clearly I should've done nothing at all instead.

**Comment by jnthn** [comment]
@duke-m There's actually a long tradition of copying chunks from IRC logs to report bugs in Perl 6 projects, and (while it's not applicable here) it often is a very good way to do it, because there are a whole bunch of bots on the IRC channel used to do evals, bisects, etc. that help identify the nature of the problem.

In this case, the issue captures both the place that apparently needs changing and the suggested change. A developer who can fix the ticket will either know enough to know what to do or have enough information to go on to find out more.

@zoffixznet I'm grateful, thanks for doing this, it would have been forgotten otherwise.

**Comment by duke-m** [comment]
Dear @zoffixznet , don't get frustrated, if you want to continue to paste IRC logs please feel free to do so.

But as you mentioned time let me try to explain what I had in mind about time when I wrote that:
Someone reporting a bug might need additional 5 to x minutes for doing it in a structured way; but now imagine 10 to n others who also might have to add 5 to x minutes each to their time investigating and understanding the bug report.

Of course it can be better to have a bug report at all than not writing it while this one report might have discovered a super-important security issue or similar. On the other hand there should be people who feel addressed and motivated to solve.

I understand that when this project started there was only a small community and folks were familiar with each other, so everyone knew what a report was about when it mentioned an IRC log. Today this project is far bigger, thanks everyone for that, including @zoffixznet

**Comment by zoffixznet** [comment]
> **[A]** Someone reporting a bug might need additional 5 to x minutes for doing it in a structured way; but now imagine **[B]** 10 to n others who also might have to add 5 to x minutes each to their time investigating and understanding the bug report.

If you deem **[B]** to be a problem, create [`.github/ISSUE_TEMPLATE.md`](https://github.com/ugexe/zef/blob/master/.github/ISSUE_TEMPLATE.md), to automatically reject all the people from **[A]** who don't have `5 to x` minutes.

**Comment by ugexe** [comment]
Take your problems with the zef issue tracker out on someone other than the guy trying to explain himself - it's not very becoming.

**Comment by zoffixznet** [comment]
> Take your problems with the zef issue tracker out on someone other than the guy trying to explain himself 

I don't have problems with zef Issue tracker. I'm using as an example to show how to use GitHub's features to resolve the problem you two have with GitHub Issues.

**Comment by duke-m** [comment]
That you're editing a lot and deleting your replies shows that this is causing bad emotions. That was not my intention I can assure you! Maybe we just have a coffee con grappa and take some deep breaths. Everything will be alright. I never meant to be rude. Only thing is, I may should have addressed this either in a new issue or by contacting you privately, but I did not knew this would cause such a trouble. I will be fine if we all just continue.

Come on... cheers? :beers:

**Comment by ugexe** [comment]
Yes, I guess that's one way to interpret it depending on how many of the deleted messages one has read in this thread.

**Comment by zoffixznet** [comment]
> That you're editing a lot and deleting your replies shows that this is causing bad emotions.

Yes, I'd prefer to leave this conversation entirely, which is why I tried closing this Issue, deleting all the content, and unsubcribing from the Issue, but people keep @zoffixznet'ting, which puts my account back into subs.
